MonoGame.Framework.Content.Pipeline/Graphics/Font/GlyphPacker.cs

```diff
@@ -7,203 +7,145 @@
 
 namespace Microsoft.Xna.Framework.Content.Pipeline.Graphics
 {
-	// Helper for arranging many small bitmaps onto a single larger surface.
-	internal static class GlyphPacker
-	{
-		public static BitmapContent ArrangeGlyphs(Glyph[] sourceGlyphs, bool requirePOT, bool requireSquare)
-		{
-			// Build up a list of all the glyphs needing to be arranged.
-			var glyphs = new List<ArrangedGlyph>();
-
-			for (int i = 0; i < sourceGlyphs.Length; i++)
-			{
-				var glyph = new ArrangedGlyph();
-
-				glyph.Source = sourceGlyphs[i];
+    // Helper for arranging many small bitmaps onto a single larger surface.
+    internal static class GlyphPacker
+    {
+        public static BitmapContent ArrangeGlyphs(GlyphData[] sourceGlyphs, bool requirePOT, bool requireSquare)
+        {
+            // Build up a list of all the glyphs needing to be arranged.
+            var glyphs = new List<ArrangedGlyph>();
+
+            for (int i = 0; i < sourceGlyphs.Length; i++)
+            {
+                var glyph = new ArrangedGlyph();
 
-				// Leave a one pixel border around every glyph in the output bitmap.
-				glyph.Width = sourceGlyphs[i].Subrect.Width + 2;
-				glyph.Height = sourceGlyphs[i].Subrect.Height + 2;
+                glyph.Source = sourceGlyphs[i];
 
-				glyphs.Add(glyph);
-			}
+                // Leave a one pixel border around every glyph in the output bitmap.
+                glyph.Width = sourceGlyphs[i].Subrect.Width + 2;
+                glyph.Height = sourceGlyphs[i].Subrect.Height + 2;
 
-			// Sort so the largest glyphs get arranged first.
-			glyphs.Sort(CompareGlyphSizes);
+                glyphs.Add(glyph);
+            }
 
-			// Work out how big the output bitmap should be.
-			int outputWidth = GuessOutputWidth(sourceGlyphs);
-			int outputHeight = 0;
+            // Sort so the largest glyphs get arranged first.
+            glyphs.Sort(CompareGlyphSizes);
 
-			// Choose positions for each glyph, one at a time.
-			for (int i = 0; i < glyphs.Count; i++)
-			{
-				PositionGlyph(glyphs, i, outputWidth);
+            // Work out how big the output bitmap should be.
+            int guessedWidth = GuessOutputWidth(sourceGlyphs);
+            var rectPacker = new MaxRectsBin(guessedWidth, 1024 * 16, GrowRule.Height);
 
-				outputHeight = Math.Max(outputHeight, glyphs[i].Y + glyphs[i].Height);
-			}
+            for (int i = 0; i < glyphs.Count; i++)
+            {
+                ArrangedGlyph glyph = glyphs[i];
+                Rectangle bounds = rectPacker.Insert(glyph.Width, glyph.Height, MaxRectsHeuristic.Bl);
```

Comment:
MaxRectsHeuristic.Bl, the "tertis" heuristic, can rotate the rectangle to fit.  This heuristic has the opportunity to lead to skewed fonts.

Reply:
Rotation is disabled when creating MaxRectsBin (5 lines above; parameter `allowRotation` is false by default)